python/cuml/test/test_pickle.py

```diff
@@ -303,6 +299,22 @@ def assert_model(pickled_model, X_test):
     pickle_save_load(tmpdir, create_mod, assert_model)
 
 
+@pytest.mark.parametrize('model_name',
+                         all_models.keys())
+def test_unfit_pickle(model_name):
+    # Any model xfailed in this test cannot be used for hyperparameter sweeps
+    # with dask or sklearn
+    if model_name in decomposition_models_xfail.keys() or \
+       model_name in ["PCA", "UMAP"]:
+        pytest.xfail()
+
+    # Pickling should work even if fit has not been called
```

Comment:
I'm trying to understand the bug in UMAP/PCA picking. Is this where they are experiencing problems?

Reply:
The issue was specifically for not-yet-fit umap/pca objects. Getstate and setstate would reference fields that hadn't been created yet. I just ensure those are set to None in the initializer.